packages/source-drupal/src/entities/Entity.js

```diff
@@ -0,0 +1,203 @@
+const axios = require('axios')
+const reduce = require('lodash.reduce')
+const uniqBy = require('lodash.uniqby')
+const { 
+  toPascalCase
+} = require('../utils')
+const slugify = require('@sindresorhus/slugify')
+
+class Entity {
+  constructor(source, { entityType, entityName, type, url }) {
+    this.source = source // instance of DrupalSource.js
+
+    this.entityType = entityType
+    this.entityName = entityName
+    this.graphQlContentType // gridsome store api addConentType
+    this.type = type // json:api type i.e. node--article
+    this.relationshipTypes = [] // array of possible relationships
+    this.url = url // url to fetch, pulled from apiSchema
+    this.response = [] // response from this.fetchData
+  }
+
+  get getEntityType() {
+    return this.entityType
+  }
+
+  get getEntityName() {
+    return this.entityName
+  }
+
+  get getType() {
+    return this.type
+  }
+
+  get getTypeName() {
+    return this.createTypeName(this.type)
+  }
+
+  createTypeName(text = '') {
```

Comment:
Other source plugins have a `typeName` option which is prefixed all types the plugin creates. This makes it easier to use Gridsome with other sources without naming collisions, and also have multiple Drupal sources. Should we do the same for this plugin? The types would then be like `DrupalNodeArticle`. And we can skip the name if it equals the entity type, so `user--user` will become `DrupalUser`. What do you think?

The store has a helper function for creating the type names. It doesn't add the `s` suffix, but we can easily add it for all source plugins later if that is desired (before 1.0 release).

```js
store.makeTypeName('node--article') >> DrupalNodeArticle
```

Reply:
Oh great! Yeah, I like that. And yes, if it equals the entity type we can skip the name. So if I'm understanding correct, another example would be `file--file`, would just be `DrupalFile`?

Good to know about the helper function - adding the `s` isn't a big deal to me, but does seem like something the people would demand :)

Reply:
Yes, I think we can just remove the last part of the string if it equals the first part for all entity types :)